### DISCUSSION TITLE
Firefly-III Helm Chart not working

### DISCUSSION THREAD

**Discussion by FortuneOSarcasm** [firefly-iii/firefly-iii#8556] [Firefly III]
I followed the instructions for installing the helm chart and everything deployed successfully but the postgres DB pod was stuck pending.

running kubectl describe pod <podname> shows that it cannot Bind the PersistentVolumeClaim

The PVC appears to be using storage class nfs-client but it has no matching PersistentVolume defined.

I manually created a PersistentVolume and redeployed the PersistentVolumeClaim so it now correctly binds and the postgresql pod starts up but the firefly-iii pod is still stuck pending.
It's failing it's health check

I see the following error in the logs for that pod

> Going to decrypt the database.
[2024-02-19 00:51:29] local.ERROR: Could not poll the database: SQLSTATE[08006] [7] could not translate host name "dbname='firefly'" to address: Name or service not known (SQL: select "id", "name", "data" from "configuration" where "name" = is_decrypted_accounts and "configuration"."deleted_at" is null limit 1)

and then a bunch of errors

> [Mon Feb 19 00:52:07.155493 2024] [php:error] [pid 405] [client 10.42.1.1:46374] PHP Fatal error:  Uncaught Illuminate\\Encryption\\MissingAppKeyException: No application encryption key has been specified. in /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php:101\nStack trace:\n#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/helpers.php(263): Illuminate\\Encryption\\EncryptionServiceProvider->Illuminate\\Encryption\\{closure}()\n#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(103): tap()\n#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(82): Illuminate\\Encryption\\EncryptionServiceProvider->key()\n#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(34): Illuminate\\Encryption\\EncryptionServiceProvider->parseKey()\n#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(873): Illuminate\\Encryption\\EncryptionServiceProvider->Illuminate\\Encryption\\{closure}()\n#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\\Container\\Container->build()\n#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\\Container\\Container->resolve()\n#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\\Foundation\\Application->resolve()\n#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\\Container\\Container->make()\n#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1423): Illuminate\\Foundation\\Application->make()\n#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(201): Illuminate\\Container\\Container->offsetGet()\n#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(188): Illuminate\\Session\\SessionManager->buildEncryptedSession()\n#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(74): Illuminate\\Session\\SessionManager->buildSession()\n#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(61): Illuminate\\Session\\SessionManager->createNativeDriver()\n#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Manager.php(105): Illuminate\\Session\\SessionManager->createFileDriver()\n#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Manager.php(80): Illuminate\\Support\\Manager->createDriver()\n#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionServiceProvider.php(52): Illuminate\\Support\\Manager->driver()\n#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(873): Illuminate\\Session\\SessionServiceProvider->Illuminate\\Session\\{closure}()\n#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\\Container\\Container->build()\n#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\\Container\\Container->resolve()\n#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\\Foundation\\Application->resolve()\n#21 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\\Container\\Container->make()\n#22 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1423): Illuminate\\Foundation\\Application->make()\n#23 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/RoutingServiceProvider.php(118): Illuminate\\Container\\Container->offsetGet()\n#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(873): Illuminate\\Routing\\RoutingServiceProvider->Illuminate\\Routing\\{closure}()\n#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\\Container\\Container->build()\n#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\\Container\\Container->resolve()\n#27 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\\Foundation\\Application->resolve()\n#28 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\\Container\\Container->make()\n#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1423): Illuminate\\Foundation\\Application->make()\n#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/RoutingServiceProvider.php(172): Illuminate\\Container\\Container->offsetGet()\n#31 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(873): Illuminate\\Routing\\RoutingServiceProvider->Illuminate\\Routing\\{closure}()\n#32 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\\Container\\Container->build()\n#33 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\\Container\\Container->resolve()\n#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\\Foundation\\Application->resolve()\n#35 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\\Container\\Container->make()\n#36 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(119): Illuminate\\Foundation\\Application->make()\n#37 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(718): app()\n#38 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(582): response()\n#39 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(490): Illuminate\\Foundation\\Exceptions\\Handler->renderHttpException()\n#40 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(356): Illuminate\\Foundation\\Exceptions\\Handler->prepareResponse()\n#41 /var/www/html/app/Exceptions/Handler.php(128): Illuminate\\Foundation\\Exceptions\\Handler->render()\n#42 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(193): FireflyIII\\Exceptions\\Handler->render()\n#43 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(170): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->renderHttpResponse()\n#44 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException()\n#45 {main}\n  thrown in /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php on line 101

Any help would be appreciated

**Comment by Bwvolleyball** [comment]
It's a half-baked chart. I can't get it to install either unfortunately.

**Reply by JC5** [reply]
Feel free to submit a PR 🙂
